libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -231,6 +253,31 @@ bool AP_ADSB::check_startup()
     return in_state.vehicle_list != nullptr;
 }
 
+
+//  detect if an instance of an ADSB sensor is connected.
+void AP_ADSB::detect_instance(uint8_t instance)
+{
+    switch (get_type(instance)) {
+    case Type::None:
+        return;
+    case Type::uAvionix_MAVLink:
+        if (AP_ADSB_uAvionix_MAVLink::detect()) {
+            _backend[instance] = new AP_ADSB_uAvionix_MAVLink(*this, instance);
+            return;
+        }
+        break;
+    }
+}
+
+// get instance type from instance
+AP_ADSB::Type AP_ADSB::get_type(uint8_t instance) const
+{
+    if (instance < ADSB_MAX_INSTANCES) {
+        return (Type)((uint8_t)_type[instance]);
```

Comment:
```suggestion
    if (instance < ARRAY_SIZE(_type)) {
        return (Type)(_type[instance].get());
```

The second line change seems to be our convention.